### ISSUE TITLE
Fallback overapproximation with an HPolytope

### ISSUE THREAD

**Issue by mforets** [JuliaReach/LazySets.jl#1691]
Compare:

```julia
julia> overapproximate(rand(Ball2), HPolytope)

MethodError: no method matching overapproximate(::Ball2{Float64}, ::Type{HPolytope})
Closest candidates are:
  overapproximate(::LazySet{N<:Real}, !Matched::Type{#s69} where #s69<:HPolygon) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Approximations/overapproximate.jl:49
  overapproximate(::LazySet{N<:Real}, !Matched::Type{#s57} where #s57<:HPolygon, !Matched::Real) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Approximations/overapproximate.jl:49
  overapproximate(::LazySet{N<:Real}, !Matched::Type{#s287} where #s287<:Interval) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Approximations/overapproximate.jl:473
  ...

Stacktrace:
 [1] top-level scope at In[19]:1
```
vs.
```julia
julia> overapproximate(rand(Ball2), HPolygon)

HPolygon{Float64}(HalfSpace{Float64,VN} where VN<:AbstractArray{Float64,1}[HalfSpace{Float64,Array{Float64,1}}([1.0, 0.0], -0.6469210534102698), HalfSpace{Float64,Array{Float64,1}}([0.0, 1.0], 0.638946226017241), HalfSpace{Float64,Array{Float64,1}}([-1.0, 0.0], 1.8615013583439128), HalfSpace{Float64,Array{Float64,1}}([0.0, -1.0], 0.575634078916402)])
```
In the former case we may just return a better error message (it may be lifted with epsilon-close approx in higher dims), or define some fallback approximation, eg. with template directions.

**Comment by schillic** [comment]
> define some fallback approximation, eg. with template directions

I disagree. The user should define what they want. "polytope overapproximation" is not well-defined. In 2D an epsilon-close approximation is feasible, so there the default is reasonable, but in high dimensions this should be requested explicitly.

**Comment by mforets** [comment]
Ok. I find it a bit uncomfortable that the following two fail:

```julia
overapproximate(rand(Ball2, dim=4), HPolytope)

overapproximate(rand(Ball2, dim=4), HPolytope, OctDirections(4))
```
Given that the following two work:

```julia
overapproximate(rand(Ball2, dim=2), HPolygon)

overapproximate(rand(Ball2, dim=2), HPolygon, 0.01)
```
Perhaps we should add a better error message for the former cases, or even allow `overapproximate(..., HPolytope, <:AbstractDirections)` by calling the existing function.

**Comment by schillic** [comment]
> overapproximate(rand(Ball2, dim=4), HPolytope)

I already commented on that. I would rather remove the default for `HPolygon` than set a default for `HPolytope`. The only reasonable default to me would be epsilon-close approximation, but this is not available yet (#969) and probably infeasible in high dimensions. Just to avoid complaints, I would say it is best to not offer a default so that the user is aware of the importance of the choice. I like the idea of defining a method that explains this in a nice error message.

> overapproximate(rand(Ball2, dim=4), HPolytope, OctDirections(4))

The `HPolytope` is redundant: just do `overapproximate(rand(Ball2, dim=4), OctDirections(4))`. `overapproximate(rand(Ball2, dim=2), 0.01)` also works. We can of course offer the method.
